code/datums/manufacturing_requirements.dm

```diff
@@ -1,216 +1,336 @@
+/datum/manufacturing_requirement/match_property
+	VAR_PROTECTED/property_id //! Material property to match by its string identifier
+	VAR_PROTECTED/property_threshold //! What threshold our property has to match or exceed in order to pass.
 
-                    PLEASE ALPHABETIZE THANKS
-***************************************************************/
+	#ifdef CHECK_MORE_RUNTIMES
+	New()
+		. = ..()
+		if (isnull(src.property_id))
+			CRASH("[src] created with null property_id")
+		if (isnull(src.property_threshold))
+			CRASH("[src] created with null property_threshold")
+	#endif
 
-/datum/manufacturing_requirement/match_property
 	is_match(var/datum/material/M)
-		var/should_match_property = !isnull(src.material_property) || !isnull(src.material_threshold)
-		if (should_match_property && !src.matches_property(M))
-			return FALSE
-		if (!isnull(src.material_flags) && !src.matches_flags(M.getMaterialFlags()))
-			return FALSE
 		. = ..()
+		if (!.) return
+		if (!src.match_property(M)) return FALSE
 
-	/// Returns whether the material flags are matched. This will return true should any flag match.
-	proc/matches_flags(var/material_flags)
-		return material_flags & src.material_flags
-
-	/// Returns whether the material property matches the given criterion. Default behavior is to check if >=, override w/o calling parent for diff behavior.
-	proc/matches_property(var/datum/material/M)
-		return M.getProperty(src.material_property) >= src.material_threshold
+	/// Returns whether the material property meets the threshold. Overwrite to have behavior other than >=
+	proc/match_property(var/datum/material/M)
+		return M.getProperty(src.property_id) >= src.property_threshold
 
 /datum/manufacturing_requirement/match_property/conductive
 	name = "Conductive"
 	id = "conductive"
-	material_property = "electrical"
-	material_threshold = 6
+	property_id = "electrical"
+	property_threshold = 6
 
 /datum/manufacturing_requirement/match_property/conductive/high
 	name = "High Energy Conductor"
 	id = "conductive_high"
-	material_threshold = 8
-
-/datum/manufacturing_requirement/match_property/crystal
-	name = "Crystal"
-	id = "crystal"
-	material_flags = MATERIAL_CRYSTAL
-
-/datum/manufacturing_requirement/match_property/crystal/dense
-	name = "Extraordinarily Dense Crystalline Matter"
-	id = "crystal_dense"
-	material_property = "density"
-	material_threshold = 7
-
-/datum/manufacturing_requirement/match_property/crystal/gemstone
-	name = "Gemstone"
-	id = "gemstone"
-
-	is_match(var/datum/material/M)
-		if (!(istype(M, /datum/material/crystal/gemstone)))
-			return FALSE
-		. = ..()
+	property_threshold = 8
 
 /datum/manufacturing_requirement/match_property/dense
 	name = "High Density Matter"
 	id = "dense"
-	material_property = "density"
-	material_threshold = 4
+	property_id = "density"
+	property_threshold = 4
 
-/datum/manufacturing_requirement/match_property/dense/super
+/datum/manufacturing_requirement/match_property/superdense
```

Comment:
Assuming flattening here for some reason? You have the nested types below still, so not like it's a consistent restructure. One or the other, right?

Reply:
Fair, changed